localstack/services/lambda_/invocation/docker_runtime_executor.py

```diff
@@ -435,13 +469,19 @@ def prepare_version(cls, function_version: FunctionVersion) -> None:
                     )
                     raise e
             if config.LAMBDA_PREBUILD_IMAGES:
-                target_path = function_version.config.code.get_unzipped_code_location()
-                prepare_image(target_path, function_version)
+                prepare_image(function_version)
 
     @classmethod
     def cleanup_version(cls, function_version: FunctionVersion) -> None:
         if config.LAMBDA_PREBUILD_IMAGES:
-            CONTAINER_CLIENT.remove_image(get_image_name_for_function(function_version))
+            # TODO re-enable image cleanup.
+            # Enabling it currently deletes image after updates as well
+            # It also creates issues when cleanup is concurrently with build
+            # probably due to intermediate layers being deleted
+            # image_name = get_image_name_for_function(function_version)
+            # LOG.debug("Removing image %s after version deletion", image_name)
+            # CONTAINER_CLIENT.remove_image(image_name)
```

Comment:
⚠️  We should follow up on this immediately because having tons of orphan prebuilt images around is annoying to clean up.

Reply:
Fully agree! As long as you do not use random names, it should not get too bad, but orphaned layers will still pollute the system.

Reply:
At the very least this needs to be fixed when it leaves its experimental stage :+1: